DemoInfo/DemoParser.cs

```diff
@@ -506,9 +527,142 @@ public void ParseHeader()
 
 			Header = header;
 
-
 			if (HeaderParsed != null)
 				HeaderParsed(this, new HeaderParsedEventArgs(Header));
+
+				if (header.PlaybackTime == 0)  //header is corrupted
+					FixHeader();
+		}
+
+		void FixHeader()
+		{
+			ExpressParseToEnd();
+			Header.PlaybackTicks = IngameTick;
+			Header.PlaybackFrames = CurrentTick - (TickAtStart + 1);
+			Header.PlaybackTime = TickInterval * IngameTick;
+			
+			CurrentTick = 0;
+
+			//the ol' switcheroo
+			//It's probably safest and easiest to
+			//swap out for a fresh bitstream.
+			Input.Seek(0, System.IO.SeekOrigin.Begin);
```

Comment:
I can try copying the stream and reading the copy for the ticks, then continuing with the initial stream for a normal demo parse.